tests/integration/framework/process/helm/helm.go

```diff
@@ -129,3 +130,30 @@ func UnmarshalStdout[K any](t *testing.T, h *Helm) []K {
 
 	return ks
 }
+
+// UnmarshalStdoutFunc unmarshals the stdout of the helm command into resources separated by "---" and calls the provided
+// function with the metadata and data of each resource so it can be unmarshaled into the appropriate type by the caller.
+func UnmarshalStdoutFunc(t *testing.T, h *Helm, f func(objMetaFound metav1.PartialObjectMetadata, data []byte)) {
```

Comment:
```suggestion
func (h *Helm) UnmarshalStdoutFunc(t *testing.T, f func(objMetaFound metav1.PartialObjectMetadata, data []byte)) {
```

Reply:
should I change the signature of the other one as well? line [107](https://github.com/dapr/dapr/pull/8230/files#diff-bc682a8a1930f631fda869caf5450fea7fccc312ca7fefaef1d29b10fb20ec54R107) above `func UnmarshalStdout[K any](t *testing.T, h *Helm) []K {`

Reply:
now, I remember, a method (like `UnmarshalStdout[K any]`) in Go cannot have generics yet

Reply:
changed